apps/design/frontend/src/app.tsx

```diff
@@ -0,0 +1,293 @@
+/* eslint-disable react/destructuring-assignment */
+/* eslint-disable @typescript-eslint/no-use-before-define */
+/* eslint-disable react/no-array-index-key */
+import './polyfills';
+import { AppBase } from '@votingworks/ui';
+import React, { useEffect, useRef, useState } from 'react';
+import { throwIllegalValue } from '@votingworks/basics';
+import {
+  AnyElement,
+  Box,
+  Color,
+  Document,
+  Ellipse,
+  GridDimensions,
+  Page,
+  Rectangle,
+} from './document_types';
+import { SvgBox, SvgEllipse, SvgPage, SvgRectangle } from './document_svg';
+import { allBubbleBallots } from './all_bubble_ballots';
+
+function replaceAtIndex<T>(array: T[], index: number, value: T): T[] {
+  return [...array.slice(0, index), value, ...array.slice(index + 1)];
+}
+
+function randomColor(): Color {
+  return `hsl(${Math.random() * 360}, 100%, 50%)`;
+}
+
+interface BaseObjectProps<T extends AnyElement> {
+  element: T;
+  setElement: (element: T) => void;
+}
+
+function RectangleObject({ element, setElement }: BaseObjectProps<Rectangle>) {
+  return (
+    <g
+      onClick={() => {
+        setElement({
+          ...element,
+          fill: randomColor(),
+        });
+      }}
```

Comment:
I imagine this color-changing on-click was just for testing purposes? Fine to leave! Just checking

Reply:
Yep, that was just to test that interactivity is possible